File: app/src/main/java/com/lmgy/exportapk/adapter/SettingsListAdapter.java
package com.lmgy.exportapk.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.lmgy.exportapk.R;
import com.lmgy.exportapk.bean.SettingsBean;

import java.util.List;

/**
 * @author lmgy
 * @date 2019/10/18
 */
public class SettingsListAdapter extends ArrayAdapter<SettingsBean> {

    private int mResource;

    public SettingsListAdapter(Context context, int resource, List<SettingsBean> objects) {
        super(context, resource, objects);
        this.mResource = resource;
    }

    @NonNull
    @Override
    @SuppressLint("ViewHolder")
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        SettingsBean bean = getItem(position);
        View view = LayoutInflater.from(getContext()).inflate(mResource, null);
        TextView textView = view.findViewById(R.id.tv_name);
        ImageView imageView = view.findViewById(R.id.iv_icon);
        textView.setText(bean.getName());
        imageView.setImageResource(bean.getImage());
        return view;
    }

}

File: app/src/main/java/com/lmgy/exportapk/adapter/SettingsListAdapter.kt
package com.lmgy.exportapk.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.lmgy.exportapk.R
import com.lmgy.exportapk.bean.SettingsContentBean

/**
 * @author lmgy
 * @date 2019/10/15
 */
class SettingsListAdapter(context: Context, private val resource: Int, objects: MutableList<SettingsContentBean>) : ArrayAdapter<SettingsContentBean>(context, resource, objects) {

    @SuppressLint("ViewHolder")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val bean = getItem(position)
        val view = LayoutInflater.from(context).inflate(resource, null)
        val textView = view.findViewById(R.id.tv_name)
        val imageView = view.findViewById(R.id.iv_icon)
        textView.setText(bean?.name)
        imageView.setImageResource(bean?.image)
        return view
    }

}